.template/variants/api/app/controllers/concerns/localization.rb

```diff
@@ -2,12 +2,24 @@ module Localization
   extend ActiveSupport::Concern
 
   included do
-    before_action :set_locale
+    around_action :switch_locale
 
     protected
 
-    def set_locale
-      I18n.locale = request.headers[:'Accept-Language'] || I18n.default_locale
+    def switch_locale(&action)
+      locale = extract_locale_from_param || I18n.default_locale
+
+      I18n.with_locale(locale, &action)
+    end
+
+    def extract_locale_from_param
+      I18n.locale_available?(params[:locale]) ? params[:locale].to_sym : nil
```

Comment:
@andyduong1920 As per our compass convention, we should follow `return` syntax instead of conditionals, can we do that here too.

Reply:
will update that @ankitkalia1195 ;)

Reply:
Fixed in https://github.com/nimblehq/rails-templates/pull/241/commits/0e06c497f7d26bb382450f349771b6564ba93356